lib/policies/load-balancing.js

```diff
@@ -319,39 +336,32 @@ TokenAwareIterator.prototype.iterate = function (callback) {
 };
 
 TokenAwareIterator.prototype.computeNext = function () {
-  //return local replicas
-  var host = this.nextLocalReplica();
-  if (host) {
-    return {value: host, done: !host};
+  var host;
+  if (this.localReplicaIndex < this.localReplicas.length) {
+    host = this.localReplicas[(this.startIndex + (this.localReplicaIndex++)) % this.localReplicas.length];
+    return { value: host, done: false };
   }
-  //return child hosts
-  var item = this.childIterator.next();
-  if (!item.done) {
+  this.startIndex = -1;
+  // Return hosts from child policy
+  var item;
+  while ((item = this.childIterator.next()) && !item.done) {
     if (this.replicaMap[item.value.address]) {
-      return this.computeNext();
+      // Avoid yielding replicas from the child load balancing policy query plan
```

Comment:
I've just pushed a fix that does the following:

1. Yield local replica hosts.
1. Delegate to child policy and filter out hosts that are not local replicas.
1. Yields the remote replicas that haven't been selected by the child policy (if any)